### ISSUE TITLE
Network interface wildcard causes segfault

### ISSUE THREAD

**Issue by maciozo** [Alexays/Waybar#392]
When I try to specify a network interface with a wildcard, a la

```
{
    "layer": "top",
    "position": "bottom",
    "height": 30,
    "modules-left": ["sway/workspaces"],
    "modules-right": ["network"],
    "network": {
        "interface": "wlp2s*",
        "format-wifi": "{essid} ({signalStrength}%) ",
        "format-ethernet": "{ifname}: {ipaddr}/{cidr} ",
        "format-linked": "{ifname} (No IP) ",
        "format-disconnected": "Disconnected ⚠",
        "format-alt": "{ifname}: {ipaddr}/{cidr}"
    }
}
```

Waybar segfaults with the following messages:
```
$ waybar --log-level trace
GLib-GIO-Message: 16:38:02.543: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
[2019-06-24 16:38:02.600] [info] Resources files: /home/maciek/.config/waybar/config, /home/maciek/.config/waybar/style.css
Unknown option --log-level
[1]    22036 segmentation fault  waybar --log-level trace
```

Waybar 0.7 on sway. Void linux kernel 5.1.12

I'll try running a build with debugging symbols

**Comment by maciozo** [comment]
```
Thread 1 "waybar" received signal SIGSEGV, Segmentation fault.
0x00005555555937b8 in waybar::modules::Network::getPreferredIface (this=<optimized out>, skip_idx=<optimized out>) at ../src/modules/network.cpp:549
549     ../src/modules/network.cpp: No such file or directory.
(gdb) backtrace
#0  0x00005555555937b8 in waybar::modules::Network::getPreferredIface (this=<optimized out>, skip_idx=<optimized out>) at ../src/modules/network.cpp:549
#1  0x000055555559af15 in waybar::modules::Network::Network (this=<optimized out>, id=..., config=...) at ../src/modules/network.cpp:101
#2  0x00005555555c4701 in waybar::Factory::makeModule (this=<optimized out>, name=...) at ../src/factory.cpp:45
#3  0x00005555555a5168 in waybar::Bar::getModules (this=<optimized out>, factory=..., pos=...) at ../src/bar.cpp:279
#4  0x00005555555a86fb in waybar::Bar::setupWidgets (this=0x5555556ff0a0) at /usr/include/c++/8.3/ext/new_allocator.h:79
#5  waybar::Bar::Bar (this=<optimized out>, w_output=<optimized out>, w_config=...) at ../src/bar.cpp:54
#6  0x00005555555a8d78 in waybar::Client::handleName (data=<optimized out>, name=<optimized out>) at /usr/include/c++/8.3/bits/unique_ptr.h:830
#7  0x00007ffff5b8fa7e in ffi_call_unix64 () from /usr/lib/libffi.so.6
#8  0x00007ffff5b8f43f in ffi_call () from /usr/lib/libffi.so.6
#9  0x00007ffff7baf2dd in ?? () from /usr/lib/libwayland-client.so.0
#10 0x00007ffff7babac9 in ?? () from /usr/lib/libwayland-client.so.0
#11 0x00007ffff7bacfb4 in wl_display_dispatch_queue_pending () from /usr/lib/libwayland-client.so.0
#12 0x00007ffff677a944 in ?? () from /usr/lib/libgdk-3.so.0
#13 0x00007ffff671aad0 in gdk_display_get_event () from /usr/lib/libgdk-3.so.0
#14 0x00007ffff677a662 in ?? () from /usr/lib/libgdk-3.so.0
#15 0x00007ffff641a52e in g_main_context_dispatch () from /usr/lib/libglib-2.0.so.0
#16 0x00007ffff641c418 in ?? () from /usr/lib/libglib-2.0.so.0
#17 0x00007ffff641c45c in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0
#18 0x00007ffff65d0ecd in g_application_run () from /usr/lib/libgio-2.0.so.0
#19 0x000055555557c655 in waybar::Client::main (this=<optimized out>, argc=<optimized out>, argv=<optimized out>) at ../src/client.cpp:278
#20 0x000055555557d583 in main (argc=1, argv=0x7fffffffe338) at ../src/main.cpp:22
```

**Comment by Alexays** [comment]
Fixed with c916fe258ef6e3b4334d9196530600372f8e7553
